micro-curator/src/main/java/com/aol/micro/server/curator/lock/DistributedLockServiceCuratorImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.aol.micro.server.curator.lock;
+
+import java.util.concurrent.TimeUnit;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.locks.InterProcessSemaphoreMutex;
+import org.apache.curator.framework.state.ConnectionState;
+import org.apache.curator.framework.state.ConnectionStateListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.aol.micro.server.utility.DistributedLockService;
+
+public class DistributedLockServiceCuratorImpl implements DistributedLockService, ConnectionStateListener {
+
+	private volatile boolean acquired;
+
+	private InterProcessSemaphoreMutex curatorLock;
```

Comment:
Better to make any fields that don't change final (and those that do volatile)

private final InterProcessSemaphoreMutex curatorLock;